app/models/concerns/host_common.rb

```diff
@@ -117,7 +117,9 @@ def crypt_root_pass
                        end
 
     if unencrypted_pass.present?
-      is_actually_encrypted = if PasswordCrypt.crypt_gnu_compatible?
+      is_actually_encrypted = if operatingsystem.try(:password_hash) == "Base64"
+                                !root_pass_changed?
```

Comment:
I don't think this is enough to stop it being re-encoded if inherited from a host group or settings, since the root_pass on Host::Managed will have changed, but it wouldn't need re-encoding if merely copied.  I think this demonstrates it:

<pre>
    test "should use hostgroup base64 root password without reencoding" do
      Setting[:root_pass] = "$1$default$hCkak1kaJPQILNmYbUXhD0"
      hg = FactoryGirl.create(:hostgroup, :with_os)
      hg.operatingsystem.update_attribute(:password_hash, 'Base64')
      hg.root_pass = "abcdefghi"
      hg.save!
      assert_equal "YWJjZGVmZ2hp", hg.root_pass

      h = FactoryGirl.create(:host, :managed, :hostgroup => hg, :operatingsystem => nil)
      h.root_pass = nil
      h.save!
      assert h.root_pass.present?
      assert_equal h.hostgroup.root_pass, h.root_pass
      assert_equal h.hostgroup.root_pass, h.read_attribute(:root_pass), 'should copy root_pass to host unmodified'
    end
</pre>